lms/djangoapps/philu_overrides/tests/test_context_processor.py

```diff
@@ -0,0 +1,590 @@
+@mock.patch('lms.djangoapps.philu_overrides.context_processor.is_org_detail_prompt_available', mock.Mock())
+@mock.patch('lms.djangoapps.philu_overrides.context_processor.get_org_metric_update_prompt', mock.Mock())
+def test_get_activation_error_alert_message_auth_failure(request_object):  # pylint: disable=redefined-outer-name
+    """
+    Test that get_global_alert_messages should NOT return activation error message as the context
+    when user is unauthenticated
+    """
+
+    request = request_object
+    request.user = AnonymousUserFactory.create()  # AnonymousUserFactory creates an unauthenticated user
+    request.user.is_active = False
+
+    expected_global_alert_message = []
+    actual_global_alert_message = get_global_alert_messages(request).get('global_alert_messages')
+
+    assert expected_global_alert_message == actual_global_alert_message
+
+
+@mock.patch('lms.djangoapps.philu_overrides.context_processor.is_org_detail_platform_overlay_available', mock.Mock())
+@mock.patch('lms.djangoapps.philu_overrides.context_processor.is_org_oef_prompt_available', mock.Mock())
+@mock.patch('lms.djangoapps.philu_overrides.context_processor.get_org_oef_update_prompt', mock.Mock())
+@mock.patch('lms.djangoapps.philu_overrides.context_processor.is_org_detail_prompt_available', mock.Mock())
+@mock.patch('lms.djangoapps.philu_overrides.context_processor.get_org_metric_update_prompt', mock.Mock())
+def test_get_activation_error_alert_message_active_user_failure(request_object):  # pylint: disable=redefined-outer-name
+    """
+    Test that get_global_alert_messages should NOT return activation error message as the context
+    when user is already active
+    """
+
+    request = request_object
+
+    expected_global_alert_message = []
+    actual_global_alert_message = get_global_alert_messages(request).get('global_alert_messages')
+
+    assert expected_global_alert_message == actual_global_alert_message
+
+
+@mock.patch('lms.djangoapps.philu_overrides.context_processor.is_org_detail_platform_overlay_available', mock.Mock())
+@mock.patch('lms.djangoapps.philu_overrides.context_processor.is_org_oef_prompt_available', mock.Mock())
+@mock.patch('lms.djangoapps.philu_overrides.context_processor.get_org_oef_update_prompt', mock.Mock())
+@mock.patch('lms.djangoapps.philu_overrides.context_processor.is_org_detail_prompt_available', mock.Mock())
+@mock.patch('lms.djangoapps.philu_overrides.context_processor.get_org_metric_update_prompt', mock.Mock())
+def test_get_activation_error_alert_message_activation_url_failure(
+    request_object  # pylint: disable=redefined-outer-name
+):
+    """
+    Test that get_global_alert_messages should NOT return activation error message as the context
+    upon direction to the activation url, i.e. when request.path contains '/activate/'
+    """
+
+    request = request_object
+    request.user.is_active = False
+    request.path = request.path + '/activate/'
```

Comment:
please use `urllib` or `urlparse` or atleast `.format`

Reply:
Please do this at all occurrences.

Reply:
Any particular reason for this? I think request.path += '/activate/' should suffice. .format is more useful IMO when you have to insert a value in the middle of a string. Here we just have to append. Thoughts?

Reply:
and using urllib or urlparse would require an extra import, no?

Reply:
Got iy, it's not exactly the url and about `.format()` it's little efficient in term of memory but I just compared the times, concatenation with `+` seems more efficient. 
<img width="517" alt="Screen Shot 2020-09-22 at 9 48 03 PM" src="https://user-images.githubusercontent.com/29256866/94245076-2fe25b00-ff33-11ea-82aa-ea221520fa34.png">